### PR TITLE
docs: add cadvisor privileged upgrade notes

### PR DESCRIPTION
**Pull request by michaellzc**
upgrade notes for https://github.com/sourcegraph/deploy-sourcegraph/pull/4126

## Test plan

docs change

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #35030** <a href="https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/35030" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sourcegraph/sourcegraph/35030?utm_source=stack-comment).

File: doc/admin/updates/kubernetes.md [hunk 1]
```diff
@@ -10,6 +10,10 @@
 
 <!-- GENERATE UPGRADE GUIDE ON RELEASE (release tooling uses this to add entries) -->
 
+## Unreleases
+
+- `cadvisor` container can collect out of memory events happening to containers and it can be used to discover underprovisoned resources. As a result, `cadvisor` now has to run in `privileged` mode. If you have your own monitoring infrastructure, you may choose to disable `cadvisor` or set `cadvisor.containerSecurityContext.privileged=false` in your override file. [#4126](https://github.com/sourcegraph/deploy-sourcegraph/pull/4126)
```
Comment (caugustus-sourcegraph) [inline doc/admin/updates/kubernetes.md:15]: Is there a reason to add this note here, versus in the Changelog? (this change also applies to docker-compose installations)

If it stays here - the text needs updating, it's still referring to helm configuration.

Comment (michaellzc) [reply doc/admin/updates/kubernetes.md]: oops...

updated both k8s and compose upgrade notes.

**Review by caugustus-sourcegraph — APPROVED** [review]
It's fine to have the notes in the upgrade docs but it still probably belongs in the changelog too.

**Comment by michaellzc** [comment]
> It's fine to have the notes in the upgrade docs but it still probably belongs in the changelog too.

🤔 I think of the [CHANGELOG.md](https://github.com/sourcegraph/sourcegraph) as product-related change and the audience is users (people who search stuff)

where the upgrade notes in docsite are stuff more relevant to operator of sourcegraph

**Pull request merged**